lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1069,32 +1069,29 @@ def run(self):
             else:
                 changed, changes, need_rebuild, force_update = new_service.compare(current_service)
                 if changed:
-                    changed = True
                     if need_rebuild:
                         if not module.check_mode:
                             self.remove_service(module.params['name'])
                             self.create_service(module.params['name'],
                                                 new_service)
                         msg = 'Service rebuilt'
                         rebuilt = True
-                        changes = changes
                     else:
                         if not module.check_mode:
                             self.update_service(module.params['name'],
                                                 current_service,
                                                 new_service)
                         msg = 'Service updated'
                         rebuilt = False
-                        changes = changes
                 else:
-                    if force_update and not module.check_mode:
-                        self.update_service(module.params['name'],
-                                            current_service,
-                                            new_service)
+                    if force_update:
+                        if module.check_mode:
```

Comment:
Shouldn't it be
`````
if force_update:
    if not module.check_mode:
`````
here?

Reply:
Indeed, my bad. I knew I should have done that _after_ sleeping.